src/routes/room.js

```diff
@@ -93,24 +90,65 @@ router.route('/add').post(async(req, res)=>{
    console.log(ID)
 
   const room = await StudyRoom.findOne( { sID: ID })
-  const participants=room.participants.push(email)
-
-   StudyRoom.updateOne(
+  var participants=room.participants;
+  if (!participants.includes(email)) {
+    participants.push(email);
+  }
+  else {
+    res.json("Student is already present in studyroom").status(200);
+    return;
+  }
+
+  await StudyRoom.updateOne(
     { sID: ID }, 
     { participants: participants },
   );
   const student = await Student.findOne( { email: email })
-  const StudyRooms=student.StudyRooms.push(sID)
+  var studyRooms=student.StudyRooms;
+  studyRooms.push(ID);
 
-  Student.updateOne(
+  await Student.updateOne(
```

Comment:
the await here has no effects

Reply:
removing the await here causes student to be undefined 
![image](https://user-images.githubusercontent.com/59633304/200646149-61a3051f-d339-4412-99d6-bb6dd4b05f5f.png)